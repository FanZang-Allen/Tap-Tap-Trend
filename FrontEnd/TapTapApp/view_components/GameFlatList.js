import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  SafeAreaView,
  FlatList,
  Image,
  RefreshControl,
} from 'react-native';

const GameFlatList = ({navigation, gotData, refreshing, onRefresh}) => {
  const Item = ({index, name, provider, avatarUrl, rating, genre, gameId}) => (
    <View style={styles.item}>
      <View style={styles.index_container}>
        <Text style={styles.game_name}>{index}</Text>
      </View>
      <View style={styles.avatar_container}>
        <Image
          source={{
            uri: avatarUrl,
          }}
          style={styles.avatar_image}
          resizeMode="contain"
        />
      </View>
      <View style={styles.text_container}>
        <Text
          style={styles.game_name}
          onPress={() => {
            navigation.navigate('Game', {game_id: gameId});
          }}>
          {name}
        </Text>
        <Text style={styles.genre_str}>{genre}</Text>
        <View style={styles.detail_container}>
          <Text style={styles.provider_name}>{provider}</Text>
          <View style={styles.rating_container}>
            <Image
              source={require('../assets/RatingStar.png')}
              style={styles.rating_image}
              resizeMode="contain"
            />
            <Text style={styles.rating_str}>{rating}</Text>
          </View>
        </View>
      </View>
    </View>
  );

  const renderItem = ({item}) => (
    <Item
      gameId={item.id}
      index={item.index}
      name={item.name}
      provider={item.provider}
      avatarUrl={item.logo}
      rating={item.rating}
      genre={item.genre}
    />
  );
  function RenderList(data) {
    return (
      <SafeAreaView style={styles.container}>
        <FlatList
          data={data}
          renderItem={renderItem}
          keyExtractor={item => item.id}
          refreshControl={
            <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
          }
        />
      </SafeAreaView>
    );
  }
  function RenderFailed() {
    return (
      <View style={styles.container}>
        <ImageBackground
          source={{
            uri: 'https://scottleejenkins.files.wordpress.com/2014/07/werwerwerwerwerwer.jpg',
          }}
          style={styles.background}
        />
      </View>
    );
  }
  function RenderLoading() {
    return (
      <View style={styles.background}>
        <Text style={styles.provider_name}>Loading</Text>
      </View>
    );
  }
  if (gotData === null) {
    return RenderLoading();
  } else if (gotData === undefined) {
    return RenderFailed();
  } else {
    return RenderList(gotData);
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  background: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 20,
    paddingBottom: 10,
  },
  item: {
    backgroundColor: '#ffffff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    borderRadius: 15,
  },
  index_container: {
    width: '7%',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignSelf: 'center',
    marginRight: 10,
  },
  text_container: {
    width: '73%',
    flex: 1,
    flexDirection: 'column',
    marginHorizontal: 16,
    justifyContent: 'space-between',
  },
  detail_container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },
  avatar_container: {
    width: '20%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  rating_container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  avatar_image: {
    flex: 1,
    width: 80,
    height: 80,
  },
  rating_image: {
    width: 20,
    height: 20,
  },
  genre_str: {
    fontSize: 18,
    color: '#9B9B9B',
    fontWeight: 'bold',
  },
  game_name: {
    fontSize: 20,
    color: 'black',
    fontWeight: 'bold',
  },
  provider_name: {
    flex: 4,
    fontSize: 14,
    color: '#9B9B9B',
  },
  rating_str: {
    flex: 4,
    fontSize: 14,
    color: '#9076FB',
  },
});

export default GameFlatList;
