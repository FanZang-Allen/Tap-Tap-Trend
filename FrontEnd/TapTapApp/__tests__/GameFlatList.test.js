import 'react-native';
import React from 'react';
import GameFlatList from '../view_components/GameFlatList';
import renderer from 'react-test-renderer';

let testData = [
  {
    genre: 'PVP · PUBG · Online',
    id: '82354',
    index: 1,
    logo: 'https://img.tapimg.com/market/icons/4cf4ff9eb0d623a2ae9cf5ecc49879c3_360.png?imageMogr2/auto-orient/strip',
    name: 'Fortnite',
    provider: 'Epic Games',
    rating: '6.9',
    region: 'Global',
  },
];

test('Game Flat List Rendered Page', () => {
  const snap = renderer
    .create(
      <GameFlatList
        gotData={testData}
        refreshing={false}
        onRefresh={() => console.log('This screen not suppport refresh')}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
