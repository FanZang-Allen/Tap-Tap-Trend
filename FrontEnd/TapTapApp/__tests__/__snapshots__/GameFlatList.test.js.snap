// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Flat List Rendered Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "genre": "PVP 路 PUBG 路 Online",
          "id": "82354",
          "index": 1,
          "logo": "https://img.tapimg.com/market/icons/4cf4ff9eb0d623a2ae9cf5ecc49879c3_360.png?imageMogr2/auto-orient/strip",
          "name": "Fortnite",
          "provider": "Epic Games",
          "rating": "6.9",
          "region": "Global",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 15,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "marginRight": 10,
                "width": "7%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              1
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "20%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://img.tapimg.com/market/icons/4cf4ff9eb0d623a2ae9cf5ecc49879c3_360.png?imageMogr2/auto-orient/strip",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": 80,
                  "width": 80,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginHorizontal": 16,
                "width": "73%",
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Fortnite
            </Text>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              PVP 路 PUBG 路 Online
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#9B9B9B",
                    "flex": 4,
                    "fontSize": 14,
                  }
                }
              >
                Epic Games
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "height": 20,
                      "width": 20,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "#9076FB",
                      "flex": 4,
                      "fontSize": 14,
                    }
                  }
                >
                  6.9
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
